### ISSUE TITLE
Mixed Precision training: How do I know if a model is supported for fp16 training.

### ISSUE THREAD

**Issue by shubhamiit** [open-mmlab/mmdetection#10830]
yolof_r50_c5_8x8_1x_coco : Worked for me with fp16 training.
While,
mask_rcnn_r50_fpn_fp16_1x_coco,
deformable_detr_twostage_refine_r50_16x2_50e_coco,
sparse_rcnn_r101_fpn_300_proposals_crop_mstrain_480-800_3x_coco,

Failed for fp16 training, How do we know if a particular model is supported or not for fp16 training?

Main Error: ->  expected scalar type Half but found Float.

/// Error: expected scalar type Half but found Float.
Traceback:
  File "swallow_all_exceptions_decorator.py", line 59, in wrapper
    return func(*args, **kwargs)
  File "finetune.py", line 782, in main
    finetune_runner(args)
  File "finetune_runner.py", line 201, in finetune_runner
    azml_trainer.train()
  File "swallow_all_exceptions_decorator.py", line 59, in wrapper
    return func(*args, **kwargs)
  File "finetune.py", line 273, in train
    self.hf_trainer = self.ft_with_trainer(
  File "finetune.py", line 327, in ft_with_trainer
    trainer.train()
  File "trainer.py", line 1664, in train
    return inner_training_loop(
  File "trainer.py", line 1940, in _inner_training_loop
    tr_loss_step = self.training_step(model, inputs)
  File "trainer.py", line 2735, in training_step
    loss = self.compute_loss(model, inputs)
  File "trainer_utils.py", line 108, in compute_loss
    return super().compute_loss(model, inputs, return_outputs=return_outputs)
  File "trainer.py", line 2767, in compute_loss
    outputs = model(**inputs)
  File "module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "distributed.py", line 1040, in forward
    output = self._run_ddp_forward(*inputs, **kwargs)
  File "distributed.py", line 1000, in _run_ddp_forward
    return module_to_run(*inputs[0], **kwargs[0])
  File "module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "model_wrapper.py", line 327, in forward
    return self.model.train_step(data, optimizer=None)
  File "base.py", line 248, in train_step
    losses = self(**data)
  File "module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "fp16_utils.py", line 119, in new_func
    return old_func(*args, **kwargs)
  File "base.py", line 172, in forward
    return self.forward_train(img, img_metas, **kwargs)
  File "sparse_rcnn.py", line 55, in forward_train
    roi_losses = self.roi_head.forward_train(
  File "sparse_roi_head.py", line 233, in forward_train
    bbox_results = self._bbox_forward(stage, x, rois, object_feats,
  File "sparse_roi_head.py", line 128, in _bbox_forward
    bbox_feats = bbox_roi_extractor(x[:bbox_roi_extractor.num_inputs],
  File "module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "fp16_utils.py", line 208, in new_func
    return old_func(*args, **kwargs)
  File "single_level_roi_extractor.py", line 100, in forward
    roi_feats_t = self.roi_layers[i](feats[i], rois_)
  File "module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "roi_align.py", line 215, in forward
    return roi_align(input, rois, self.output_size, self.spatial_scale,
  File "roi_align.py", line 95, in forward
    ext_module.roi_align_forward(

//

Also,
is it true that by adding this :
fp16 = dict(loss_scale=512.)
By adding above line, we can make a model fp16 compatible?

**Comment by hhaAndroid** [comment]
@shubhamiit In principle, all models are capable of training with Automatic Mixed Precision (AMP). If a model doesn't support it, it means that it hasn't been modified yet. Community users can provide feedback or assistance to help fix the issue.
